AntennaTracker/control_auto.cpp

```diff
@@ -32,23 +35,32 @@ void Tracker::update_auto(void)
         update_yaw_servo(bf_yaw);
     }
 }
-void Tracker::calc_angle_error( float pitch, float yaw)
+void Tracker::calc_angle_error( float pitch, float yaw, bool direction_reversed)
 {
-	// Pitch angle error in centidegrees
-	// Positive error means the target is above current pitch
+    // Pitch angle error in centidegrees
+    // Positive error means the target is above current pitch
     // Negative error means the target is below current pitch
     float ahrs_pitch = ahrs.pitch_sensor;
     nav_status.angle_error_pitch = pitch - ahrs_pitch;
 
     // Yaw angle error in centidegrees
-	// Positive error means the target is right of current yaw
+    // Positive error means the target is right of current yaw
     // Negative error means the target is left of current yaw
     int32_t ahrs_yaw_cd = wrap_180_cd(ahrs.yaw_sensor);
     nav_status.angle_error_yaw = wrap_180_cd(yaw - ahrs_yaw_cd);
+    if(direction_reversed == true){
+    	if (nav_status.angle_error_yaw>0){
+    	nav_status.angle_error_yaw = (yaw - ahrs_yaw_cd) - 36000;
+    	}
+    	else if (nav_status.angle_error_yaw<=0){
```

Comment:
No need for this extra check of angle_error_yaw.  the precending "if" ensures that it's <= 0.